core/thread/inc/ThreadPool.h

```diff
@@ -0,0 +1,106 @@
+#ifndef THREADPOOL_H_
+#define THREADPOOL_H_
+
+#include "tbb/tbb.h"
+#include "TPool.h"
+
+class ThreadPool: public TPool<ThreadPool> {
+public:
+   explicit ThreadPool(){
+      this->initTBB->initialize();
+   }
+
+   explicit ThreadPool(size_t nThreads){
+      this->initTBB->initialize(nThreads);
+   }
+
+   ~ThreadPool() {
+      this->initTBB->terminate();
+   }
+
+
+   template<class F> auto Map(F func, unsigned nTimes) -> std::vector<decltype(func())>;
+   /// \cond
+   template<class F, class INTEGER> auto Map(F func, ROOT::TSeq<INTEGER> args) -> std::vector <decltype(func(args.front()))>;
+   template<class F, class T> auto Map(F func, std::vector<T> &args) -> std::vector<decltype(func(args.front()))>;
+   // / \endcond
+   using TPool<ThreadPool>::Map;
+
+   template<class T, class R> auto Reduce(const std::vector<T> &objs, R redfunc) -> decltype(redfunc(objs.front(), objs.front()));
+   using TPool<ThreadPool>::Reduce;
+
+private:
+   tbb::task_scheduler_init *initTBB = new tbb::task_scheduler_init(tbb::task_scheduler_init::deferred);
```

Comment:
This is sub-optimal.  This means that this files can only be used when TBB is available (i.e. TThreadPool becomes an optional (and likely underused) part of ROOT.).   Is TBB really necessary here and can't it be 'abstracted' away?